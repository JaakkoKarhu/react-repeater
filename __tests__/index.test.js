import React from 'react';
import { shallow } from 'enzyme';

import Repeater from '../src/index';

test('Basic rendering test', () => {

  const repeater = shallow(
    <Repeater />
  )
  expect(repeater).toMatchSnapshot()
})

test('Returns mapped data to passed callback function onChange', () => {
  // Not sure if this test is a bit silly
  let receivedData = {}
  const onChange = (e, data) => {
    receivedData = data
  }
  const repeater = shallow(
    <Repeater>
      <input dataKey='test-key'
             onChange={ onChange } />
    </Repeater>
  )
  repeater.find('input').simulate('change', { target: { value: 'a' }})
  expect(receivedData[0]['test-key']).toBe('a')
})

test('Maps passed data to inputs', () => {
  const value = 'This is mapped value.',
        data = [{ 'test-key': value }]
  const repeater = shallow(
    <Repeater data={ data }>
      <input dataKey='test-key' />
    </Repeater>
  )
  const input = repeater.find('input')
  expect(input.node.props.value).toBe(value)
})