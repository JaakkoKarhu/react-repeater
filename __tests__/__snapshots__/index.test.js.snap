// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic rendering test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="repeater"
>
    <div
        className="repeater-add"
        onClick={[Function]}
    />
</div>,
  "nodes": Array [
    <div
      className="repeater"
>
      <div
            className="repeater-add"
            onClick={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Repeater />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Repeater {
        "_reactInternalInstance": [Circular],
        "add": [Function],
        "componentDidMount": [Function],
        "context": Object {},
        "getElems": [Function],
        "onChange": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "dataValues": Array [
            Object {},
          ],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="repeater"
>
          <div
                    className="repeater-add"
                    onClick={[Function]}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="repeater"
>
          <div
                    className="repeater-add"
                    onClick={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Repeater />,
}
`;
